### ISSUE TITLE
Youtube Ciphered Videos Not Able To Download

### ISSUE THREAD

**Issue by praburamWAPKA** [Athlon1600/youtube-downloader#10]
Hi , if you are true coder then just try to find ciphered video download link should work 

for example try this video link it wont able to download shows you 404 Error

https://api.unblockvideos.com/youtube_downloader?id=QQ4Gds8phUk&selector=mp4

So better try to find decryption function to download all videos like that i mentioned above and already youtube-dl software using this decryption of ciphered videos so try your best

**Comment by mattjis** [comment]
It's because of the IP. Only the IP that extracts the VEVO video's url can view it. Try extracting the URL on your own IP.. I've tested and it works fine for me.

**Comment by praburamWAPKA** [comment]
my ip looks like 2405:204:738a:3868:a0b3:7c04:506e:6471

and its not working

**Comment by praburamWAPKA** [comment]
I just tried everything its not working....
if you have a code And just Try it

**Comment by Athlon1600** [comment]
does the video URL for that video generated via python's youtube-dl working for you?

**Comment by praburamWAPKA** [comment]
No they are using  js player to generate link ... well they are using Similar function like Youtube-dl....
if we use youtube-dl it took too long to generate link because its youtube-dl need shell access for each and every Urls :)

**Comment by praburamWAPKA** [comment]
youtube-dl have Decrypting Function of youtube ciphered videos like vevo....bla,bla..

**Comment by praburamWAPKA** [comment]
yes youtube-dl function works fine and i have created one project based on this 
https://master-land.net/Thread-Youtube-API-Source-PHP-Script-For-Youtube-Download-Links

**Comment by praburamWAPKA** [comment]
But problem is have to use youtube-dl.... And it will become slow to show an output ... because it need shell command for each request will down server as well

**Comment by chadul** [comment]
same to me, return 403

**Comment by hex-firebone** [comment]
Hello guys,

why don't you guys try this library

https://github.com/hex-firebone/YouTube-Downloader
